src/effects/VST3/VST3Wrapper.cpp

```diff
@@ -653,13 +647,21 @@ void VST3Wrapper::ConsumeChanges(const EffectSettings& settings)
 //
 //As a side effect subsequent call to IAudioProcessor::process may flush
 //plugin internal buffers
-void VST3Wrapper::FlushParameters(EffectSettings& settings)
+void VST3Wrapper::FlushParameters(EffectSettings& settings, bool* hasChanges)
 {
    if(!mActive)
    {
       auto componentHandler = static_cast<ComponentHandler*>(mComponentHandler.get());
       componentHandler->FlushCache(settings);
 
+      const auto doProcessing = !GetSettings(settings).parameterChanges.empty();
+
+      if(hasChanges != nullptr)
+         *hasChanges = doProcessing;
+
+      if(!doProcessing)
+         return;
```

Comment:
I see a new early exit, which also eliminates some unnecessary work, besides fixing the issue described.  That's good